seed/analysis_pipelines/pipeline.py

```diff
@@ -156,28 +160,78 @@ def _stop_task_chain(task_instance):
                 _stop_task_chain(_self)
                 return
             except Exception as e:
+                # yikes, we didn't account for this. Try to bow out as gracefully as possible
+                reraise_exception = False
                 try:
-                    Analysis.objects.get(id=_analysis_id)
-
+                    with transaction.atomic():
+                        # don't wait for lock to avoid deadlock
+                        locked_analysis = (
+                            Analysis.objects
+                            .select_for_update(nowait=True)
+                            .get(id=_analysis_id)
+                        )
+
+                        # analysis still exists, and we were able to get a lock on it
+                        # something unexpected happened during the task
+                        if locked_analysis.in_terminal_state():
+                            # someone else has already finished the task off, just log and exit gracefully
+                            log_message = {
+                                'analysis_id': _analysis_id,
+                                'debug_message': 'Task raised unhandled exception, but the analysis was already in a terminal state. '
+                                                 'Ignoring the exception and stopping the celery task chain.',
+                                'exception': repr(e)
+                            }
+                            logger.info(json.dumps(log_message))
+                            _stop_task_chain(_self)
+                            return
+                        else:
+                            # *force* the analysis to the failed status, log messages, and raise the exception
+                            locked_analysis.status = Analysis.FAILED
+                            locked_analysis.save()
+                            AnalysisMessage.log_and_create(
+                                logger=logger,
+                                type_=AnalysisMessage.ERROR,
+                                user_message='Unexpected error occurred.',
+                                debug_message='Caught unexpected exception occurred during analysis task and the analysis still exists. '
+                                              'Failing the analysis and re-raising the exception.',
+                                analysis_id=_analysis_id,
+                                analysis_property_view_id=None,
+                                exception=e,
+                            )
```

Comment:
not using `AnalysisPipeline.fail()` here b/c that method tries to grab a lock on the analysis as well. From my testing locally it shouldn't matter (a nested transaction can lock and access same row as a parent) but avoiding it still b/c it makes me uncomfortable.